packages/backend/src/nest/registration/registration.service.ts

```diff
@@ -1,178 +1,68 @@
-import { Inject, Injectable, OnModuleInit } from '@nestjs/common'
-import express from 'express'
-import getPort from 'get-port'
-import { Agent, Server } from 'http'
+import { Injectable, OnModuleInit } from '@nestjs/common'
 import { EventEmitter } from 'events'
-import {
-  registerOwner,
-  registerUser,
-  RegistrarResponse,
-  RegistrationResponse,
-  sendCertificateRegistrationRequest,
-} from './registration.functions'
-import {
-  ConnectionProcessInfo,
-  ErrorCodes,
-  ErrorMessages,
-  LaunchRegistrarPayload,
-  PermsData,
-  RegisterOwnerCertificatePayload,
-  SocketActionTypes,
-} from '@quiet/types'
-import { EXPRESS_PROVIDER } from '../const'
+import { extractPendingCsrs, issueCertificate } from './registration.functions'
+import { ErrorCodes, ErrorMessages, PermsData, RegisterOwnerCertificatePayload, SocketActionTypes } from '@quiet/types'
 import { RegistrationEvents } from './registration.types'
-import { ServiceState } from '../connections-manager/connections-manager.types'
 import Logger from '../common/logger'
 
 @Injectable()
 export class RegistrationService extends EventEmitter implements OnModuleInit {
   private readonly logger = Logger(RegistrationService.name)
-  private _server: Server
-  private _port: number
-  public registrationService: any
   public certificates: string[] = []
   private _permsData: PermsData
-  private _ownerCertificate: string
 
-  constructor(@Inject(EXPRESS_PROVIDER) public readonly _app: express.Application) {
+  constructor() {
     super()
   }
 
   onModuleInit() {
-    this.on(RegistrationEvents.SET_CERTIFICATES, certs => {
-      this.setCertificates(certs)
-    })
-    this.on(RegistrationEvents.REGISTER_USER_CERTIFICATE, async (csr: string) => {
-      if (!this._permsData) {
-        console.log('NO PERMS DATA')
-        return
-      }
-      await this.registerUser(csr)
-    })
-    this.setRouting()
-  }
-
-  public setCertificates(certs: string[]) {
-    this.certificates = certs
-  }
-
-  private pendingPromise: Promise<RegistrarResponse> | null = null
-
-  private setRouting() {
-    // @ts-ignore
-    this._app.use(express.json())
-    this._app.post('/register', async (req, res): Promise<void> => {
-      if (this.pendingPromise) return
-      this.pendingPromise = this.registerUser(req.body.data)
-      const result = await this.pendingPromise
-      if (result) {
-        res.status(result.status).send(result.body)
+    this.on(
+      RegistrationEvents.REGISTER_USER_CERTIFICATE,
+      async (payload: { csrs: string[]; certificates: string[] }) => {
+        // Lack of permsData means that we are not the owner of the community
+        if (!this._permsData) return
```

Comment:
Also I think it would be good to move this into `issueCertificates` to keep all the logic together.